+ Карточка персонажа

karvalolka opened the pull request.

Commit:
+ Карточка персонажа

kovdmit commented on resources/views/profile/index.blade.php:
```diff
@@ -0,0 +1,52 @@
+                                    <div class="card-body p-4">
+                                        <div class="text-center">
+                                            <h5 class="fw-bolder">{{ $char->nickname }}</h5>
+                                            <p>{{$char->race->name}}</br></p>
```
Эта конструкция используется несколько раз в коде ниже. Предлагаю создать для персонажа метод получения названия расы, фракции, класса.

kovdmit commented on resources/views/profile/index.blade.php:
```diff
@@ -0,0 +1,52 @@
+                                        <div class="text-center">
+                                            <h5 class="fw-bolder">{{ $char->nickname }}</h5>
+                                            <p>{{$char->race->name}}</br></p>
+                                            <p>{{$char->grade->name}}</br></p>
```
См. Коммент выше

kovdmit commented on resources/views/profile/index.blade.php:
```diff
@@ -0,0 +1,52 @@
+                                            <h5 class="fw-bolder">{{ $char->nickname }}</h5>
+                                            <p>{{$char->race->name}}</br></p>
+                                            <p>{{$char->grade->name}}</br></p>
+                                            <p>{{$char->fraction->name}}</br></p>
```
См коммент выше

kovdmit commented on resources/views/profile/show.blade.php:
```diff
@@ -0,0 +1,53 @@
+                    <table>
+                        <tr>
+                            <td style="width: 100px;">Раса:</td>
+                            <td>{{ $char->race->name }}</td>
```
См коммент выше

kovdmit commented on resources/views/profile/show.blade.php:
```diff
@@ -0,0 +1,53 @@
+                        </tr>
+                        <tr>
+                            <td>Грейд:</td>
+                            <td>{{ $char->grade->name }}</td>
```
См коммент выше

kovdmit commented on resources/views/profile/show.blade.php:
```diff
@@ -0,0 +1,53 @@
+                        </tr>
+                        <tr>
+                            <td>Фракция:</td>
+                            <td>{{ $char->fraction->name }}</td>
```
См коммент выше

Commit:
- profile/index.blade.php

Commit:
+ profile/index.blade.php;
+ методы для получения названий расы, фракции и класса

Commit:
+ лимит создания персонажей (2);
+ колонки в таблице chars для "надевания" оружия;
+ "надевание" оружия;
+ отображение урона с учётом ндетого оружия

kovdmit commented on app/Http/Controllers/Admin/Char/StoreCharController.php:
```diff
@@ -13,6 +13,12 @@ public function __invoke(StoreRequest $request)
         $data['user_id'] = auth()->id();
+        $characterCount = Char::where('user_id', $data['user_id'])->count();
+
+        if ($characterCount >= 2) {
```
Необходимо избавиться от магических числел

kovdmit commented on app/Http/Controllers/Personal/Char/StoreCharController.php:
```diff
@@ -13,6 +13,12 @@ public function __invoke(StoreRequest $request)
     {
         $data = $request->validated();
         $data['user_id'] = auth()->id();
+        $characterCount = Char::where('user_id', $data['user_id'])->count();
```
Дублируется бизнес-логика.

kovdmit commented on app/Models/Weapon.php:
```diff
@@ -16,4 +16,8 @@ public function attackRate()
     {
         return $this->belongsTo(AttackRate::class, 'attack_rates_id');
     }
+    public function chars()
```
Одно оружие может быть у нескольких персонажей?

kovdmit commented on database/migrations/2024_10_14_172403_add_column_weapon_id_in_chars_table.php:
```diff
@@ -0,0 +1,53 @@
+     */
+    public function down(): void
+    {
+        Schema::create('chars_new', function (Blueprint $table) {
```
Что тут происходит вообще?

karvalolka replied:
особенность sqlite в том что в таблицах нельзя  удалять ключи. Создаётся новая таблица, импортируются данные из старой, старая удаляется и переименовывается

kovdmit replied:
Это ответ не на мой вопрос.

karvalolka replied:
создаётся таблица с новым неймом и старыми столбцами.
в неё сохраняются данные из действующей таблицы,
старая таблица удаляется, новая ренеймится

kovdmit commented on resources/views/main/index.blade.php:
```diff
@@ -34,7 +34,7 @@
                                             <p style="text-align: center;">
                                                 <td class="col-2">Атака</td>
-                                                <td>{{$char->attack_power}}</td>
+                                                <td>{{$char->attack_power + ($characterWeapons[$char->id]->power ?? 0)}}</td>
```
Надо вынести в отдельный метод

kovdmit commented on resources/views/profile/show.blade.php:
```diff
@@ -38,16 +37,35 @@
                             <td>Атака:</td>
-                            <td>{{ $char->attack_power }}</td>
+                            <td>
+                                @if ($weapon)
```
Вынести в отдельный метод

kovdmit commented on resources/views/profile/show.blade.php:
```diff
@@ -38,16 +37,35 @@
                         </tr>
+                        <tr>
+                            <td>Оружие:</td>
+                            <td>{{ $char->weapon ? $char->weapon->name : 'Нет оружия' }}</td>
```
Вынести в отдельный метод

kovdmit commented on app/Http/Controllers/Admin/Char/StoreCharController.php:
```diff
@@ -13,6 +13,12 @@ public function __invoke(StoreRequest $request)
     {
```
В очередной раз повторяю: не надо коммитить новую разработку в существующий PR.
Если ты скидываешь PR на ревью, а я написал замечания, то надо их исправить, потом слить ветки, а потом только создавать новый PR и пушить новые изменения.
Либо вести разработку каждой фичи в отдельной ветке.

Commit:
- магические числа;
- дублирование логики в админовском и пользовательском сторконтроллерах;
изменено отношение оружия к персонажу

Commit:
изменён редирект

Commit:
изменён редирект;
починил связь оружия и персонажа
